shared-libs/lineage/src/hydration.js

```diff
@@ -21,6 +21,42 @@ const extractParentIds = current => selfAndParents(current)
   .map(parent => parent._id)
   .filter(id => id);
 
+const getId = (item) => item && (typeof item === 'string' ? item : item._id);
+
+const getContactById = (contacts, id) => id && contacts.find(contact => contact && contact._id === id);
+
+const getContactIds = (contacts) => {
+  const ids = [];
+  contacts.forEach(doc => {
+    if (!doc) {
+      return;
+    }
+
+    const id = getId(doc.contact);
+    id && ids.push(id);
+
+    if (!validLinkedDocs(doc)) {
+      return;
+    }
+    Object.keys(doc.linked_docs).forEach(key => {
+      const id = getId(doc.linked_docs[key]);
+      id && ids.push(id);
+    });
+  });
+
+  return _.uniq(ids);
+};
+
+// don't process linked docs for reports
+// linked_docs property should be a key-value object
+const validLinkedDocs = doc => {
+  if (!doc || doc.type === 'data_record' || !doc.linked_docs) {
```

Comment:
I circled around this a lot in my mind: whether to allow links on reports. 
And I did, for a while, and I changed this on Friday, after realizing that I'm passing `context.contact` in message-utils (you also added a question about that). That meant that reports wouldn't have access to linked_docs anyway, so it would just be strange for them to be there, but not be usable. 

I'm not sure what the best approach would be, so it's clean and intuitive. 
What do you think?

Reply:
I think it's find to not allow linked docs on reports for now. We can easily add it if and when we have a use case that needs it.

For now I think a readability fix would be to rewrite this to check for contact types only (the old school hardcoded types and `contact`).